src/NHibernate/AdoNet/AbstractBatcher.cs

```diff
@@ -316,6 +316,8 @@ public virtual void CloseCommands()
 
 		private void CloseCommand(DbCommand cmd)
 		{
+			if (cmd == null)
+				return;
```

Comment:
Shouldn't it be `cmd?.Dispose();`?

Reply:
The net effect of the null ref exception was to discard all the method remaining code. Doing `cmd?.Dispose();` will have side effects that were not done previously in case of `null`.
If going that way, then I would also change the abstract entity persister for having it no more calling close with null. And I think all current calls to close should be checked for similar cases, in order to avoid an unwanted side effect.

Reply:
Yes, I did not notice the return.